packages/@lwc/engine-core/src/patches/detect-non-standard-aria.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import features from '@lwc/features';
+import { defineProperty, getOwnPropertyDescriptor, isNull, isUndefined } from '@lwc/shared';
+import { onReportingEnabled, report, ReportingEventId } from '../framework/reporting';
+import { LightningElement } from '../framework/base-lightning-element';
+import { logWarnOnce } from '../shared/logger';
+import { getAssociatedVMIfPresent, VM } from '../framework/vm';
+import { BaseBridgeElement } from '../framework/base-bridge-element';
+
+//
+// The goal of this code is to detect usages of non-standard reflected ARIA properties. These are caused by
+// legacy non-standard Element.prototype extensions added by the @lwc/aria-reflection package.
+//
+
+// See the README for @lwc/aria-reflection
+const NON_STANDARD_ARIA_PROPS = [
+    'ariaActiveDescendant',
+    'ariaControls',
+    'ariaDescribedBy',
+    'ariaDetails',
+    'ariaErrorMessage',
+    'ariaFlowTo',
+    'ariaLabelledBy',
+    'ariaOwns',
+];
+
+function isLightningElement(elm: Element) {
+    // The former case is for `this.prop` (inside component) and the latter is for `element.prop` (outside component).
+    // In both cases, we apply the non-standard prop even when the global polyfill is disabled, so this is kosher.
+    return elm instanceof LightningElement || elm instanceof BaseBridgeElement;
+}
+
+function findVM(elm: Element): VM | undefined {
+    // If it's a shadow DOM component, then it has a host
+    const { host } = elm.getRootNode() as ShadowRoot;
+    const vm = isUndefined(host) ? undefined : getAssociatedVMIfPresent(host);
+    if (!isUndefined(vm)) {
+        return vm;
+    }
+    // Else it might be a light DOM component. Walk up the tree trying to find the owner
+    let parentElement: Element | null = elm;
+    while (!isNull((parentElement = parentElement.parentElement))) {
+        if (isLightningElement(parentElement)) {
+            const vm = getAssociatedVMIfPresent(parentElement);
+            if (!isUndefined(vm)) {
+                return vm;
+            }
+        }
+    }
```

Comment:
This is potentially expensive, perf-wise, but I don't think there are many light DOM components out there. Anyway, the time we spend here will be limited by the depth of the tree. (And most likely we won't have to walk far to get to the component root anyway.)